File: FixMyCar/FixMyCar.Model/Entities/CarRepairShopService.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.Entities
{
    public class CarRepairShopService
    {
        public int Id { get; set; }
        [ForeignKey(nameof(CarRepairShop))]
        public int CarRepairShopId { get; set; }
        public CarRepairShop CarRepairShop { get; set; }
        [ForeignKey(nameof(ServiceType))]
        public int ServiceTypeId { get; set; }
        public ServiceType ServiceType { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }
        public double Discount { get; set; }
        public double DiscountedPrice { get; set; }
        public string State { get; set; }
        public byte[]? ImageData { get; set; }
        public string? Details { get; set; }
        public TimeSpan Duration { get; set; }
    }
}

File: FixMyCar/FixMyCar.UI/fixmycar_client/lib/src/models/car_repair_shop_service/car_repair_shop_service.dart
import 'package:json_annotation/json_annotation.dart';

part 'car_repair_shop_service.g.dart';

@JsonSerializable()
class CarRepairShopService {
  int id;
  int serviceTypeId;
  String serviceTypeName;
  String name;
  double price;
  double discount;
  double discountedPrice;
  String state;
  String? imageData;
  String? details;
  String duration;

  CarRepairShopService(
      this.id,
      this.serviceTypeId,
      this.serviceTypeName,
      this.name,
      this.price,
      this.discount,
      this.discountedPrice,
      this.state,
      this.imageData,
      this.details,
      this.duration);

  factory CarRepairShopService.fromJson(Map<String, dynamic> json) =>
      _$CarRepairShopServiceFromJson(json);

  Map<String, dynamic> toJson() => _$CarRepairShopServiceToJson(this);
}
